vizro-core/src/vizro/models/_components/ag_grid.py

```diff
@@ -77,12 +83,33 @@ class AgGrid(VizroBaseModel):
         Field(default=[]),  # TODO[MS]: here and elsewhere: do we need to validate default here?
     ]
 
-    _input_component_id: str = PrivateAttr()
+    _inner_component_id: str = PrivateAttr()
     _validate_figure = field_validator("figure", mode="before")(validate_captured_callable)
 
+    def model_post_init(self, context) -> None:
+        super().model_post_init(context)
+        self._inner_component_id = self.figure._arguments.get("id", f"__input_{self.id}")
+
     @property
     def _action_outputs(self) -> dict[str, _IdProperty]:
-        return {"__default__": f"{self.id}.children"}
+        # TODO-AV2 E: Implement _action_trigger where makes sense.
+        #  For the AgGrid the mapping could look like: {"__default__": f"{self._inner_component_id}.cellClicked"}
```

Comment:
Please can you move this somewhere outside the `_action_outputs` method? Otherwise it makes me think like something needs to change in this function.